From a35068843537b0f43ab6394a6ba39c58fb0b6b68 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 12 Oct 2018 11:07:47 +0300
Subject: [PATCH 4854/5242] MLK-19021: drm/imx/dcss: fix RTRM clock reference

commit  80344cc36ebb861722277bc9dbd511ce12a45df6 from
https://source.codeaurora.org/external/imx/linux-imx.git

RTRM clock reference was saved in apb_clk, instead of rtrm_clk. Hence,
when blanking and clocks go off, APB clock counter was 2, instead of 1.

Because IRQ_STEER controller uses APB clock as well, the APB clock ref
counter will never go to 0. Unless DCSS is never used.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-common.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dcss/dcss-common.c b/drivers/gpu/imx/dcss/dcss-common.c
index fa76c9c..704bf4b 100644
--- a/drivers/gpu/imx/dcss/dcss-common.c
+++ b/drivers/gpu/imx/dcss/dcss-common.c
@@ -276,7 +276,7 @@ static int dcss_clks_init(struct dcss_soc *dcss)
 		{"axi",   &dcss->axi_clk},
 		{"pix_div", &dcss->pdiv_clk},
 		{"pix_out", &dcss->pout_clk},
-		{"rtrm",  &dcss->apb_clk},
+		{"rtrm",  &dcss->rtrm_clk},
 		{"dtrc",  &dcss->dtrc_clk},
 	};
 
-- 
1.7.9.5

